pandas/core/frame.py

```diff
@@ -1388,6 +1388,10 @@ def to_gbq(self, destination_table, project_id=None, chunksize=None,
             or string contents. This is useful for remote server
             authentication (eg. Jupyter/IPython notebook on remote host).
 
+        Returns
+        -------
+        None
```

Comment:
IIRC the validator should detect when there is no return and not require it to be explicitly stated

Reply:
There is actually an [empty return statement](https://github.com/pydata/pandas-gbq/blob/5e80908770f46b8512ddc1c4ec192befd33d818d/pandas_gbq/gbq.py#L1151) .

Reply:
Right but it still shouldn't be required (see #25008) - was this giving you an error?

Reply:
Yes, it gives an error on my local.

Reply:
Oh OK I see. Is this any different from the other io methods? This usually is None or str though totally familiar with pandas_gbq internals.

If it never actually returns anything probably better to update the code to not return instead of adding cruft to the docstring

Reply:
It just returns to terminate the function in case given DataFrame is empty. So yeah it only returns None. 

There are also other functions (See [to_hdf](https://github.com/pandas-dev/pandas/pull/26234/files#diff-03b380f521c43cf003207b0711bac67fR2396), [to_pickle](https://github.com/pandas-dev/pandas/pull/26234/files#diff-03b380f521c43cf003207b0711bac67fR2625), [assert_series_equal](https://github.com/pandas-dev/pandas/pull/26234/files#diff-9d7323cc6b0cfa3f4482bc8d4268966dR1013)). These are functions returning a function either with empty return statement or doesn’t return.

Reply:
Thanks for confirming. Specifically with the latter there is already a PR in progress to remove that (see #25462)

I think we should do that for the other items here as well rather than adding to the docstring. Would be good as pre-cursors (separate PRs) to this

Reply:
So I will remove the return sections of docstrings for to_gbq, to_hdf and to_pickle in this branch. Then open another PR for removing return statements of these functions. Right?

Reply:
Probably best to open the other PR before continuing here

Reply:
Right thanks

Reply:
I think detecting all the docstrings with Returns set to `None` is trivial. Personally I'm happy to leave this as it is until we fix the code that detects whether something can be detected (so we can validate `RT01` in the CI), and we can easily detect and remove them later.

Reply:
I guess this issue will be resolved after merging #25462. None returning io functions are already handled in #26238. I think I can remove None since there will be no RT01 type errors after merge.

Reply:
@ihsansecer those PRs should be on master now - can you merge master and fix these up?

Reply:
@WillAyd done